### PR TITLE
12/ClientGUI - Network API

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implement the GUI Network API.

You can find all the necessary function inside. Refere to the RFC for more informations.

### CODE REVIEW

**Review by GuyomT — APPROVED** [review]
What a PR :exploding_head: 

LGTM

File: src/zappy_gui_src/network/api/get/retcode.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/// List of all the possible retcodes
+static const retcodes_t retcodes[] = {
+    {200, 0, EXPECTED},
+    {201, 1, EXPECTED},
+    {202, 1, EXPECTED},
+    {402, 1, EXPECTED},
+    {501, 1, EXPECTED},
+    {502, 1, EXPECTED},
+    {-1, -1, EXPECTED}
```
Comment (osvegn) [inline src/zappy_gui_src/network/api/get/retcode.c:25]: what are these values ?

File: src/zappy_gui_src/network/api/authentification.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+/// \file src/zappy_gui_src/network/api/authentification.c
+
+#include "intern_data.h"
+#include "client_utils.h"
+#include "api.h"
+#include <stdlib.h>
+#include <unistd.h>
+
+zappy_client_t *client_data = NULL;
```
Comment (31Nathan) [inline src/zappy_gui_src/network/api/authentification.c:16]: Documentation

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Missing Tests, otherwise good

**Review by osvegn — CHANGES_REQUESTED** [review]

**Pull request merged**